remove '.' symbol from filename (bug 38430)

jdlrobson opened the pull request:
replace any dot characters with a - to avoid file verification error

jrobson committed:
remove '.' symbol from filename (bug 38430)

replace any dot characters with a - to avoid file verification error

yuvipanda commented:
Should account for everything from https://www.mediawiki.org/wiki/Manual:$wgLegalTitleChars

jdlrobson commented:
Good thinking! Will do so today. Interestingly this doesn't seem to make reference to . as being an illegal character but as I understand it does list #<>[]|{} as illegal.

jrobson committed:
deal with other invalid file name characters

separate Monument function from monuments.js and make available to tests
write tests to test filename generation

jdlrobson commented:
done with tests. Please check I caught them all..
